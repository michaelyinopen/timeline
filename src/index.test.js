import React from 'react';
import MyComponent from './index.js';
import renderer from 'react-test-renderer';

test('Can create component', () => {
  const component = renderer.create(
    <MyComponent />,
  );
});