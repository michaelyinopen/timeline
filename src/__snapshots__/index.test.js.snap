// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="timeline__time-container"
>
  <div
    className="timeline__group-axis"
  >
    <ol
      className="timeline__list--none-style"
    >
      <li>
        <div
          className="timeline__group-header-wrapper"
          style={
            Object {
              "height": "nullpx",
            }
          }
        >
          <div
            aria-describedby={null}
            className="timeline__group-header"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Machine 1"
          >
            M1
          </div>
        </div>
      </li>
      <li>
        <div
          className="timeline__group-header-wrapper"
          style={
            Object {
              "height": "nullpx",
            }
          }
        >
          <div
            aria-describedby={null}
            className="timeline__group-header"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Machine 2"
          >
            M2
          </div>
        </div>
      </li>
      <li>
        <div
          className="timeline__group-header-wrapper"
          style={
            Object {
              "height": "nullpx",
            }
          }
        >
          <div
            aria-describedby={null}
            className="timeline__group-header"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Machine 3"
          >
            M3
          </div>
        </div>
      </li>
      <li>
        <div
          className="timeline__group-header-wrapper"
          style={
            Object {
              "height": "nullpx",
            }
          }
        >
          <div
            aria-describedby={null}
            className="timeline__group-header"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title="Machine 4"
          >
            M4
          </div>
        </div>
      </li>
    </ol>
  </div>
  <div
    className="timeline__slide-container timeline__schedule-container"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={0}
  >
    <div />
    <div
      className="timeline__schedule-slide"
      onMouseDown={[Function]}
      style={
        Object {
          "transform": "translate(0px,0px)",
          "width": 0,
        }
      }
    >
      <ol
        className="timeline__groups timeline__list--none-style"
      >
        <li>
          <div
            className="timeline__group-wrapper"
          >
            <div />
            <div
              className="timeline__group-content"
            >
              <ol
                className="timeline__items timeline__list--none-style"
                style={
                  Object {
                    "height": 28,
                  }
                }
              >
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J1
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J2
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J3
                    </div>
                  </div>
                </li>
              </ol>
            </div>
          </div>
        </li>
        <li>
          <div
            className="timeline__group-wrapper"
          >
            <div />
            <div
              className="timeline__group-content"
            >
              <ol
                className="timeline__items timeline__list--none-style"
                style={
                  Object {
                    "height": 28,
                  }
                }
              >
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J2
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J1
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J3
                    </div>
                  </div>
                </li>
              </ol>
            </div>
          </div>
        </li>
        <li>
          <div
            className="timeline__group-wrapper"
          >
            <div />
            <div
              className="timeline__group-content"
            >
              <ol
                className="timeline__items timeline__list--none-style"
                style={
                  Object {
                    "height": 28,
                  }
                }
              >
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J2
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J1
                    </div>
                  </div>
                </li>
              </ol>
            </div>
          </div>
        </li>
        <li>
          <div
            className="timeline__group-wrapper"
          >
            <div />
            <div
              className="timeline__group-content"
            >
              <ol
                className="timeline__items timeline__list--none-style"
                style={
                  Object {
                    "height": 28,
                  }
                }
              >
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J2
                    </div>
                  </div>
                </li>
                <li>
                  <div
                    className="timeline__item-wrapper"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "transform": "translate(0px, 0px)",
                        "width": "0px",
                      }
                    }
                  >
                    <div />
                    <div
                      className="timeline__item-content"
                    >
                      Item: 
                      J3
                    </div>
                  </div>
                </li>
              </ol>
            </div>
          </div>
        </li>
      </ol>
    </div>
  </div>
  <div
    className="timeline__time-axis timeline__slide-container"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onWheel={[Function]}
  >
    <div
      className="timeline__time-axis-slide"
      onMouseDown={[Function]}
      style={
        Object {
          "transform": "translate(0px,0px)",
          "width": 0,
        }
      }
    >
      <svg
        className="timeline__time-axis-svg"
      >
        <line
          className="timeline__time-axis-line"
          x1="0%"
          x2="100%"
          y1="0"
          y2="0"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <line
          className="timeline__minor-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="8px"
        />
        <line
          className="timeline__major-tick-mark"
          x1="0px"
          x2="0px"
          y1="0"
          y2="16px"
        />
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:00
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:10
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:20
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:30
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:40
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          00:50
        </text>
        <text
          alignmentBaseline="hanging"
          dx="2px"
          dy="2px"
          fill="black"
          x="0px"
        >
          01:00
        </text>
      </svg>
    </div>
    <ol
      className="timeline__time-axis-labels timeline__list--none-style"
    >
      <li>
        <div
          className="timeline__time-axis-label-dot"
          style={
            Object {
              "transform": "translateX(0px) translate(-50%, -50%)",
            }
          }
        />
        <div
          className="timeline__time-axis-label-time-connection"
          style={
            Object {
              "transform": "translateX(0px) translateX(-50%)",
            }
          }
        />
        <div
          className="timeline__time-axis-time-label"
          style={
            Object {
              "transform": "translateX(0px) translateX(0px)",
              "width": "48px",
            }
          }
        >
          00:00
        </div>
      </li>
      <li>
        <div
          className="timeline__time-axis-label-dot"
          style={
            Object {
              "transform": "translateX(0px) translate(-50%, -50%)",
            }
          }
        />
        <div
          className="timeline__time-axis-label-time-connection"
          style={
            Object {
              "transform": "translateX(0px) translateX(-50%)",
            }
          }
        />
        <div
          className="timeline__time-axis-time-label"
          style={
            Object {
              "transform": "translateX(0px) translateX(0px)",
              "width": "48px",
            }
          }
        >
          01:00
        </div>
      </li>
    </ol>
  </div>
</div>
`;
